include/revng/PipelineC/PipelineC.h

```diff
@@ -0,0 +1,382 @@
+#include "revng/Pipeline/Target.h"
+#include "revng/Pipes/PipelineManager.h"
+#else
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#endif
+
+#ifdef __cplusplus
+typedef Pipeline::PipelineManager rp_runner;
+#else
+typedef struct rp_runner rp_runner;
+#endif
+
+#ifdef __cplusplus
+typedef const Pipeline::Kind rp_kind;
+#else
+typedef struct rp_kind rp_kind;
+#endif
+
+#ifdef __cplusplus
+typedef Pipeline::Step rp_step;
+#else
+typedef struct rp_step rp_step;
+#endif
+
+#ifdef __cplusplus
+typedef Pipeline::ContainerBase rp_container_checkpoint;
+#else
+typedef struct rp_container_checkpoint rp_container_checkpoint;
+#endif
+
+#ifdef __cplusplus
+typedef const Pipeline::ContainerFactorySet::value_type rp_container;
+#else
+typedef struct rp_container rp_container;
+#endif
+
+#ifdef __cplusplus
+typedef const Pipeline::Target rp_target;
+#else
+typedef struct rp_target rp_target;
+#endif
+
+#ifdef __cplusplus
+typedef const Pipeline::Structure::State rp_runner_status;
+#else
+typedef struct rp_runner_status rp_runner_status;
+#endif
+
+#ifdef __cplusplus
+typedef const Pipeline::Structure::State::value_type rp_step_status;
+#else
+typedef struct rp_step_status rp_step_status;
+#endif
+
+#ifdef __cplusplus
+typedef const Pipeline::ContainersStatus::value_type rp_container_status;
+#else
+typedef struct rp_container_status rp_container_status;
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Every pointer returned by this library is owned by the library and must not
+ * be deleted, except for pointers returned by functions containing the word
+ * "create". Objects returned by functions named "create" are owned by the
+ * caller and must be destroyed invoking the appropriate function called
+ * rp_TYPE_destroy.
+ */
+
+/**
+ * Must be invoked before any other rp_* function is used,
+ * must be invoked exactly once.
+ * This will take care of initializing all llvm related
+ * stuff needed by the revng pipeline to operate.
+ *
+ * Do not initialize all stuff on your own.
+ */
+bool rp_init(int argc,
+             char *argv[],
+             int libraries_count,
+             const char *libraries_path[]);
+
+/**
+ * pipeline_path, flags, execution_folder cannot be null.
+ * execution_folder can be empty, if it is empty then the content of the
+ * pipeline will not be loaded and saved on disk before and after the execution.
+ * Flags can be empty, FlagsCount must be the size of the Flags array.
+ *
+ * returns the created rp_runner if no error happened, nullptr otherwise.
+ *
+ * this function can be called only once, since it will take take of the
+ * initialization of all dynamically loaded libraries.
+ *
+ * this must be invoked after rp_init and all invocations of
+ * rp_load_library_permanently
+ */
+rp_runner *rp_runner_create(uint64_t pipelines_count,
+                            const char *pipelines_path[],
+                            uint64_t FlagsCount,
+                            const char *Flags[],
```

Comment:
Just flags sounds a bit confusing, if we say they are *pipeline* flags it's a bit better.
```suggestion
                            uint64_t pipeline_flags_count,
                            const char *pipeline_flags[],
```